atef/widgets/config/run.py

```diff
@@ -0,0 +1,285 @@
+    return widget
+
+
+def combine_results(results: List[Result]) -> Result:
+    """
+    Combines results into a single result.
+
+    Takes the highest severity, and currently all the reasons
+    """
+
+    severity = util.get_maximum_severity([r.severity for r in results])
+    reason = str([r.reason for r in results]) or ''
+
+    return Result(severity=severity, reason=reason)
+
+
+class RunCheck(DesignerDisplay, QWidget):
+    """
+    Widget to be added to run widgets
+
+    Connections: (to establish)
+    - next button
+    - verify button to pop-out and record
+
+    Parent widget must be a PageWidget
+    """
+    filename = 'run_check.ui'
+
+    # Left to right
+    status_label: QLabel
+    status_verify_spacer: QSpacerItem
+    verify_button: QToolButton
+    verify_run_spacer: QSpacerItem
+    run_button: QPushButton
+    run_next_spacer: QSpacerItem
+    next_button: QPushButton
+
+    style_icons = {
+        Severity.success: QStyle.SP_DialogApplyButton,
+        Severity.warning : QStyle.SP_TitleBarContextHelpButton,
+        Severity.internal_error: QStyle.SP_DialogCancelButton,
+        Severity.error: QStyle.SP_DialogCancelButton
+    }
+
+    unicode_icons = {
+        # check mark
+        Severity.success: '<span style="color: green;">&#10004;</span>',
+        Severity.warning : '<span style="color: orange;">?</span>',
+        # x mark
+        Severity.internal_error: '<span style="color: red;">&#10008;</span>',
+        Severity.error: '<span style="color: red;">&#10008;</span>',
+    }
+
+    def __init__(self, *args, configs=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        icon = self.style().standardIcon(self.style_icons[Severity.warning])
+        self.status_label.setPixmap(icon.pixmap(25, 25))
+        self.configs = configs
+
+        if configs:
+            self.setup_buttons(configs=configs)
+            # initialize tooltip
+            self.update_status_label_tooltip()
+
+    def setup_buttons(self, configs, next_widget: AtefItem = None) -> None:
+        """
+        Wire up buttons to the provided config dataclass.
+        Run results and verification information will be saved to the
+        provided ``config`` dataclass.
+        For passive checkouts this should be a Prepared variant.
+
+        Link Run button to the .run() or .compare() method of config
+        Link Next button of previous widget to this widget's parent
+        Link Verify button to verify method if it exists.  If not
+        remove button and spacer.
+        Link Status to result of Run procedure
+        """
+        self._make_run_slot(configs)
+        if next_widget:
+            self.setup_next_button(next_widget)
+
+        self.setup_verify_button()
+
+    def infer_step_type(self, config: Union[PreparedComparison, ProcedureStep]) -> str:
+        if hasattr(config, 'compare'):
+            return 'passive'
+        elif hasattr(config, 'run'):
+            return 'active'
```

Comment:
Is there a more direct way to do this?

Reply:
I'll leave a TODO.  I wasn't immediately satisfied with just checking against step types, since as we add new steps we'd have to remember to maintain that list.  I'll circle back to this later